### DISCUSSION TITLE
Installation via ``pip``

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#150] [Ideas]
_Based on issue #145_

> python -m pip install "pyrx @ git+https://github.com/CEXT-Dan/PyRx.git#subdirectory=pyrx"
> python -m pip uninstall pyrx

It works! but it means that the user will install the package from the latest commit. A better way is to tag the release with a version tag (``git tag v.2.1.2``). Then the user could ``python -m pip install "pyrx @ git+https://github.com/CEXT-Dan/PyRx.git#subdirectory=pyrx@v.2.1.2"``. An even better way is to tag the commit with a version tag and a tag ``latest``, then the user could ``python -m pip install "pyrx @ git+https://github.com/CEXT-Dan/PyRx.git#subdirectory=pyrx@latest"``

```bash
git tag v.2.1.2
git tag -f latest
git push origin --tag -f
```

Additionally I think ``setup.py`` should be placed in the root directory and ``pyproject.toml`` should be added, but all this is already done here: https://github.com/gswifort/PyRx/tree/73-cad-bundle-installer
We can try to merge this later or you can use this and adjust it manually.

**Comment by CEXT-Dan** [comment]
I created a tag, latest didn't work,
this did
python -m pip install "pyrx @ git+https://github.com/CEXT-Dan/PyRx.git@v2.1.2#subdirectory=pyrx"

**Reply by gswifort** [reply]
What does it mean that `latest` doesn't work? I tested it on a temporary repo, but I added the tag via git, in the console, not via github.

**Reply by CEXT-Dan** [reply]
```
PS C:\Users\Dan> python -m pip install "pyrx @ git+https://github.com/CEXT-Dan/PyRx.git@latest#subdirectory=pyrx"
Collecting pyrx@ git+https://github.com/CEXT-Dan/PyRx.git@latest#subdirectory=pyrx
  Cloning https://github.com/CEXT-Dan/PyRx.git (to revision latest) to c:\users\dan\appdata\local\temp\pip-install-8wjwgk3_\pyrx_11ba5c179593450398f95f3944c02942
  Running command git clone --filter=blob:none --quiet https://github.com/CEXT-Dan/PyRx.git 'C:\Users\Dan\AppData\Local\Temp\pip-install-8wjwgk3_\pyrx_11ba5c179593450398f95f3944c02942'
  WARNING: Did not find branch or tag 'latest', assuming revision or ref.
  Running command git checkout -q latest
  error: pathspec 'latest' did not match any file(s) known to git
  error: subprocess-exited-with-error

  × git checkout -q latest did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git checkout -q latest did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Reply by gswifort** [reply]
You need to add `latest` tag to the same commit as 2.1.2 (and any future ones).

**Reply by CEXT-Dan** [reply]
there's a check box when I create the release.
I want to work through GitHub, since this is going to fire the workflow to PyPi

![latest](https://github.com/user-attachments/assets/fdb76292-9f9a-4226-892e-a0e5aaccb58a)

**Comment by CEXT-Dan** [comment]
If setup is in root, does pip checkout the whole repository?

I’m thinking about trying to setup a workflow to work with PyPi
https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

**Reply by gswifort** [reply]
In setup.py you can (should) specify the source directory.
I did it in `pyproject.toml`
[tool.setuptools.packages.find]
include = ["pyrx*"]

**Reply by CEXT-Dan** [reply]
I will look at your pyproject.toml again, I had forgot about it.  I probably should have looked first. 
Also, I just stuck everything in one directory. I have everything uninstalled when I'm developing, it's simpler for me to swap out one stub folder in vscode

**Reply by CEXT-Dan** [reply]
I moved setup.py to root, it works :)
I played around with checking out the latest, couldn't get it to work . not really a big deal, I will add the link in each release so users can copy paste
